### PR TITLE
New PDF certificates

### PR DESCRIPTION
**Pull request by Thiemann96**
Had to open a new PR because i forked from develop. #276 was the old PR

 - QR code is generated in the backend 
 - PDFCreator class which can be reused in other components and which

### CODE REVIEW

File: apps/mainsite/badge_pdf.py [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+                competency_name = competencies[i]['name']
+                competency = competency_name
+                #   competency = (competency_name[:35] + '...') if len(competency_name) > 35 else competency_name
+                rounded_rect = RoundedRectFlowable(0, -10, 515, 45, 10, text=competency, strokecolor="#492E98", fillcolor="#F5F5F5", studyload= studyload, esco=competencies[i]['escoID'])    
+                Story.append(rounded_rect)
+                Story.append(Spacer(1, 10))   
+                    
+                if esco: 
+                    Story.append(Spacer(1, 10))
+                    text_style = ParagraphStyle(name='Text_Style',fontName="Rubik-Italic", fontSize=10, leading=15.6, alignment=TA_CENTER, leftIndent=-35, rightIndent=-35)
```
Comment (jona159) [inline apps/mainsite/badge_pdf.py:234]: Same as above

File: apps/mainsite/badge_pdf.py [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+
+    def add_description(self, first_page_content, description):
+        description_style = ParagraphStyle(name='Description', fontSize=12,fontName='Rubik-Regular',  leading=16.5, alignment=TA_CENTER, leftIndent=20, rightIndent=20)
+        first_page_content.append(Paragraph(description, description_style))
+        first_page_content.append(Spacer(1, 10))
+
+
+    def add_narrative(self, first_page_content, narrative):
+        if narrative is not None:
+            narrative_style = ParagraphStyle(name='Narrative',fontName="Rubik-Italic",fontSize=12,textColor='#6B6B6B', leading=16.5, alignment=TA_CENTER, leftIndent=20, rightIndent=20)
```
Comment (jona159) [inline apps/mainsite/badge_pdf.py:86]: ```Rubik-Italic``` is not a registered font yet, can you add it or use a different font?

**Review by jona159 — CHANGES_REQUESTED** [review]
I have some findings: 
 - Please add or replace the missing font.
 - I think the ```qrcode``` module is missing in the ```requirements.txt```? 
 - When the description of the badge is not very long, the qr code section is not rendered at the bottom of the page but more in the middle
 - Have you tried awarding badges with competencies? I think there is something broken, for me it does not show all competencies on the pdf

File: apps/mainsite/badge_pdf.py (hunk 2 above)
Comment (Thiemann96) [reply apps/mainsite/badge_pdf.py:94]: just added it

**Comment by jona159** [comment]
The esco text is now included for every competency, it should only be displayed once at the bottom of the last page

escokompetenzen

**Review by jona159 — APPROVED** [review]
@Thiemann96 can be merged, but please remove the comments / only keep them when necessary

**Pull request merged**